src/android/usb_host/device_enumerator.cpp

```diff
@@ -0,0 +1,164 @@
+    jstring jACTION_USB_PERMISSION = env->NewStringUTF("com.librealuvc.USBHOST_PERMISSION");
+    jobject intentObject = env->NewObject(intentClass, intentCtorId, jACTION_USB_PERMISSION);
+    jclass pendingIntentClass = env->FindClass("android/app/PendingIntent");
+    jmethodID getBroadcastMethodId = env->GetStaticMethodID(pendingIntentClass, "getBroadcast", "(Landroid/content/Context;ILandroid/content/Intent;I)Landroid/app/PendingIntent;");
+    jobject permissionIntentObject = env->CallStaticObjectMethod(pendingIntentClass, getBroadcastMethodId, appObject, 0, intentObject, 0);
+
+    // Get the objects needed to iterate over the hash map
+    jclass hashMapClass = env->FindClass("java/util/HashMap");
+    jmethodID hashMapValuesMethodID = env->GetMethodID(hashMapClass, "values", "()Ljava/util/Collection;");
+    jobject deviceListCollection = env->CallObjectMethod(deviceList, hashMapValuesMethodID);
+
+    jclass collectionClass = env->FindClass("java/util/Collection");
+    jmethodID collectionIteratorMethodID = env->GetMethodID(collectionClass, "iterator", "()Ljava/util/Iterator;");
+    jobject collectionIteratorObject = env->CallObjectMethod(deviceListCollection, collectionIteratorMethodID);
+
+    jclass iteratorClass = env->FindClass("java/util/Iterator");
+    jmethodID iteratorHasNextMethodID = env->GetMethodID(iteratorClass, "hasNext", "()Z");
+    jmethodID iteratorNextMethodID = env->GetMethodID(iteratorClass,  "next", "()Ljava/lang/Object;");
+    
+    jclass usbDeviceClass = env->FindClass("android/hardware/usb/UsbDevice");
+    jmethodID usbDeviceVendorIdMethodID = env->GetMethodID(usbDeviceClass, "getVendorId", "()I");
+    jmethodID usbDeviceProductIdMethodID = env->GetMethodID(usbDeviceClass, "getProductId", "()I");
+    jmethodID usbDeviceNameMethodID = env->GetMethodID(usbDeviceClass, "getDeviceName", "()Ljava/lang/String;");
+    jmethodID usbManagerOpenDeviceMethodID = env->GetMethodID(usbManagerClass, "openDevice", "(Landroid/hardware/usb/UsbDevice;)Landroid/hardware/usb/UsbDeviceConnection;");
+    jmethodID usbManagerHasPermissionMethodID = env->GetMethodID(usbManagerClass, "hasPermission", "(Landroid/hardware/usb/UsbDevice;)Z");
+    jclass usbDeviceConnectionClass = env->FindClass("android/hardware/usb/UsbDeviceConnection");
+    jmethodID usbDeviceConnectionGetFileDescMethodID = env->GetMethodID(usbDeviceConnectionClass, "getFileDescriptor", "()I");
+    jmethodID requestPermissionMethodId = env->GetMethodID(usbManagerClass, "requestPermission", "(Landroid/hardware/usb/UsbDevice;Landroid/app/PendingIntent;)V");
+
+    // Iterate over the device list
+    while((bool)env->CallBooleanMethod(collectionIteratorObject, iteratorHasNextMethodID)) {
+      jobject device = env->CallObjectMethod(collectionIteratorObject, iteratorNextMethodID);
+      
+      // Rigel
+      // device.getVendorId() == 0x2936
+      // device.getProductId() == 0x1202
```

Comment:
> 
> 
> > > Use forks for public repo please.
> > 
> > 
> > What benefit does this bring?
> > > Instead of copy-pasting, could you keep the history for usbhost from Realsense (and import the one from Google for usbhost.{c,h}) ?
> > 
> > 
> > Yes it would have been nice to be able to do it this way, but the two forks (intel's and ours) have diverged so much that we can't reuse Intels commits without amending them as it just won't build anymore. You'd have to go through [each of these](https://github.com/IntelRealSense/librealsense/commits/fd6ccc78829653adc6b3b9252239fa69ba05d98e/src/android) and a) remove the bits we don't need and b) make sure it builds and still works on all platforms.
> 
>     1. Save bandwith of CI/people cloning librealuvc and avoids the 'I forgot to delete the branch after merge'
> 
>     2. My attempt would have to been to export those patches and filter them but since it is not trivial I don't mind not doing.
>        However, could you split in the commits the new files from the one we modified/created ?